Mathlib/Analysis/Asymptotics/ReflectsGrowth.lean

```diff
@@ -0,0 +1,267 @@
+
+lemma isTheta_one_right (h : ReflectsGrowthAddMul f g l) {a} : f a =Θ[l] (1 : α → ℝ) ↔ g a = 0 :=
+  h.reflectsGrowth₀.isTheta_one_right
+
+lemma isBigO_one_right (h : ReflectsGrowthAddMul f g l) {a} : f a =O[l] (1 : α → ℝ) ↔ g a ≤ 0 :=
+  h.reflectsGrowth₀.isBigO_one_right
+
+lemma isBigO_one_left (h : ReflectsGrowthAddMul f g l) {a} : (1 : α → ℝ) =O[l] f a ↔ 0 ≤ g a :=
+  h.reflectsGrowth₀.isBigO_one_left
+
+lemma isLittleO_one_right (h : ReflectsGrowthAddMul f g l) {a} : f a =o[l] (1 : α → ℝ) ↔ g a < 0 :=
+  h.reflectsGrowth₀.isLittleO_one_right
+
+lemma isLittleO_one_left (h : ReflectsGrowthAddMul f g l) {a} : (1 : α → ℝ) =o[l] f a ↔ 0 < g a :=
+  h.reflectsGrowth₀.isLittleO_one_left
+
+lemma isBoundedUnder_le_norm (h : ReflectsGrowthAddMul f g l) {a} :
+    IsBoundedUnder (· ≤ ·) l (‖f a ·‖) ↔ g a ≤ 0 :=
+  h.reflectsGrowth₀.isBoundedUnder_le_norm
+
+lemma tendsto_zero (h : ReflectsGrowthAddMul f g l) {a} : Tendsto (f a) l (𝓝 0) ↔ g a < 0 :=
+  h.reflectsGrowth₀.tendsto_zero
+
+lemma tendsto_norm_atTop (h : ReflectsGrowthAddMul f g l) {a} :
+    Tendsto (‖f a ·‖) l atTop ↔ 0 < g a :=
+  (isLittleO_one_left_iff _).symm.trans h.isLittleO_one_left
+
+lemma comp_tendsto {β : Type*} {l' : Filter β} {u : β → α} [NeBot l']
+    (hl : ReflectsGrowthAddMul f g l) (hu : Tendsto u l' l) :
+    ReflectsGrowthAddMul (fun x y ↦ f x (u y)) g l' where
+  eventuallyEq_norm_map_add a b := hu.eventually (hl.eventuallyEq_norm_map_add a b)
+  tendsto_of_pos hpos := (hl.tendsto_of_pos hpos).comp hu
+  isBigO_of_eq_zero hzero := (hl.isBigO_of_eq_zero hzero).comp_tendsto hu
+  eventually_ne := hu.eventually hl.eventually_ne
+
+lemma mono (hl : ReflectsGrowthAddMul f g l) {l'} [NeBot l'] (hle : l' ≤ l) :
+    ReflectsGrowthAddMul f g l' := hl.comp_tendsto hle
+
+end ReflectsGrowthAddMul
+
+lemma reflectsGrowthAddMul_zpow_cobounded {𝕜 : Type*} [NontriviallyNormedField 𝕜] :
+    ReflectsGrowthAddMul (fun (n : ℤ) (x : 𝕜) ↦ x ^ n) (AddMonoidHom.id ℤ) (cobounded 𝕜) where
+  eventuallyEq_norm_map_add _ _ := (eventually_ne_cobounded 0).mono fun x hx ↦ by
+    simp only [zpow_add₀ hx, norm_mul]
+  tendsto_of_pos h := by
+    simpa only [norm_zpow] using (tendsto_zpow_atTop_atTop h).comp tendsto_norm_cobounded_atTop
+  isBigO_of_eq_zero _ := by simp_all [isBoundedUnder_const]
+  eventually_ne := (eventually_ne_cobounded 0).mono fun _ ↦ zpow_ne_zero _
+
+lemma reflectsGrowthAddMul_zpow_atTop :
+    ReflectsGrowthAddMul (fun (n : ℤ) (x : ℝ) ↦ x ^ n) (AddMonoidHom.id ℤ) atTop :=
+  reflectsGrowthAddMul_zpow_cobounded.mono (by simp)
+
+lemma reflectsGrowthAddMul_zpow_atBot :
+    ReflectsGrowthAddMul (fun (n : ℤ) (x : ℝ) ↦ x ^ n) (AddMonoidHom.id ℤ) atBot :=
+  reflectsGrowthAddMul_zpow_cobounded.mono (by simp)
+
+@[simp]
+lemma isLittleO_zpow_zpow_atTop {m n : ℤ} : (· ^ m : ℝ → ℝ) =o[atTop] (· ^ n) ↔ m < n :=
+  reflectsGrowthAddMul_zpow_atTop.isLittleO
+
+lemma reflectsGrowth₀_pow_atTop : ReflectsGrowth₀ (fun (n : ℕ) (x : ℝ) ↦ x ^ n) (.id _) atTop := by
```

Comment:
Why is this statement different to the `zpow` one? Why do you need all three of `ReflectsGrowth`, `ReflectsGrowth_0`, `ReflectsGrowthAddMul`?